### ISSUE TITLE
map editor doesn't work with wxWidgets 3.0

### ISSUE THREAD

**Issue by filux** [MegaGlest/megaglest-source#76]
You can compile it and launch without errors but nothing more. 
Looks like map area is even missing, and in this place is visible toolbar's background.

```
[ Debian ] [ testing ] [ stretch ] [ x86_64 ]
[5170.02fe61e]
Using wxWidgets version [3.0.2.0]
```

![mapeditor](https://cloud.githubusercontent.com/assets/5335953/10398733/f220d392-6eae-11e5-8431-ee1d1183e461.png)

**Comment by softcoder** [comment]
Should be fixed in sdl2 branch

**Comment by filux** [comment]
for me on 8499d25 this doesn't work still but I dug a bit deeper and with Debug build using gcc 5.2.1 I see something like this during start up:

```
./megaglest_editor 
Using wxWidgets version [3.0.2.0]
18:08:42: Debug: Unsupported OpenGL attribute 2080637296
18:08:42: Debug: Unsupported OpenGL attribute 21946
../src/unix/glx11.cpp(50): assert "fbc" failed in wxGLContext(): invalid GLXFBConfig for OpenGL
../src/gtk/window.cpp(3454): assert "m_widget != __null" failed in SetFocus(): invalid window
../src/gtk/window.cpp(2963): assert "(m_widget != __null)" failed in DoGetClientSize(): invalid window
Trace/breakpoint trap
``` 
```
ASSERT INFO:
../src/unix/glx11.cpp(50): assert "fbc" failed in wxGLContext(): invalid GLXFBConfig for OpenGL

BACKTRACE:
[1] wxGLContext::wxGLContext(wxGLCanvas*, wxGLContext const*)
[2] MapEditor::GlCanvas::GlCanvas(MapEditor::MainWindow*, wxWindow*, int*)
[3] MapEditor::MainWindow::MainWindow(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
[4] MapEditor::App::OnInit()
[5] wxAppConsoleBase::CallOnInit()
[6] wxEntry(int&, wchar_t**)
[7] main
[8] __libc_start_main
```
```
ASSERT INFO:
../src/gtk/window.cpp(3454): assert "m_widget != __null" failed in SetFocus(): invalid window

BACKTRACE:
[1] wxWindow::SetFocus()
[2] MapEditor::MainWindow::MainWindow(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
[3] MapEditor::App::OnInit()
[4] wxAppConsoleBase::CallOnInit()
[5] wxEntry(int&, wchar_t**)
[6] main
[7] __libc_start_main
```
... with normal build using clang 3.5.2 I see this:
```
./megaglest_editor 
18:15:02: Warning: Mismatch between the program and library build versions detected.
The library used 3.0 (wchar_t,compiler with C++ ABI 1009,wx containers,compatible with 2.8),
and your program used 3.0 (wchar_t,compiler with C++ ABI 1002,wx containers,compatible with 2.8).
Using wxWidgets version [3.0.2.0]
```
... and with Debug build using clang I see this:
```
./megaglest_editor 
18:17:09: Warning: Mismatch between the program and library build versions detected.
The library used 3.0 (wchar_t,compiler with C++ ABI 1009,wx containers,compatible with 2.8),
and your program used 3.0 (wchar_t,compiler with C++ ABI 1002,wx containers,compatible with 2.8).
Using wxWidgets version [3.0.2.0]
../src/common/menucmn.cpp(760): assert "item" failed in Check(): wxMenu::Check: no such item
Trace/breakpoint trap
```
editor is starting successfully (clang + Debug) and I can even e.g. open map but any choose on toolbar ("grass" button in this case) = this:
```
ASSERT INFO:
../src/common/menucmn.cpp(760): assert "item" failed in Check(): wxMenu::Check: no such item

BACKTRACE:
[1] wxMenuBase::Check(int, bool)
[2] MapEditor::MainWindow::uncheckBrush()
[3] MapEditor::MainWindow::onMenuBrushSurface(wxCommandEvent&)
[4] wxAppConsoleBase::CallEventHandler(wxEvtHandler*, wxEventFunctor&, wxEvent&) const
[5] wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&)
[6] wxEventHashTable::HandleEvent(wxEvent&, wxEvtHandler*)
[7] wxEvtHandler::TryHereOnly(wxEvent&)
[8] wxEvtHandler::ProcessEventLocally(wxEvent&)
[9] wxEvtHandler::ProcessEvent(wxEvent&)
[10] wxWindowBase::TryAfter(wxEvent&)
[11] wxWindowBase::TryAfter(wxEvent&)
[12] wxEvtHandler::SafelyProcessEvent(wxEvent&)
[13] wxToolBarBase::OnLeftClick(int, bool)
[14] g_signal_emit_valist
[15] g_signal_emit_by_name
[16] g_signal_emit_valist
[17] g_signal_emit
[18] g_closure_invoke
[19] g_signal_emit_valist
[20] g_signal_emit
[21] g_closure_invoke
[22] g_signal_emit_valist
[23] g_signal_emit
[24] gtk_propagate_event
[25] gtk_main_do_event
[26] g_main_context_dispatch
[27] g_main_loop_run
[28] gtk_main
[29] wxGUIEventLoop::DoRun()
[30] wxEventLoopBase::Run()
[31] wxAppConsoleBase::MainLoop()
[32] MapEditor::App::MainLoop()
[33] wxEntry(int&, wchar_t**)
[34] main
[35] __libc_start_main
```

... especially this abi mismatch from clang looks interesting and gently shows that maybe something is not 100% fine with precompiled wxWidgets 3.0 on Debian.

**Comment by softcoder** [comment]
you need to delete build folder and re-run cmake and build

**Comment by softcoder** [comment]
Tested in ubuntu and debian Sid and works fine

**Comment by filux** [comment]
fixed before 3.12.0 release
